lib/src/layer/polyline_layer.dart

```diff
@@ -58,8 +53,13 @@ class Polyline {
 @immutable
 class PolylineLayer extends StatelessWidget {
   final List<Polyline> polylines;
+  @Deprecated('has no effect anymore, polyline culling is enabled by default and controlled by margin')
   final bool polylineCulling;
 
+  /// extent outside of the viewport before culling polylines, set to null to
+  /// disable polyline culling
+  final double? polylineCullingMargin;
```

Comment:
Does this need to be customizable? Can we foresee any issues with this? I'd be in favour of just removing this as well as the `polylineCulling`. I realise that the option is needed to properly support gradients, but maybe it's worthwhile just turning it off behind the scenes if a gradient is specified, and keeping it on otherwise. I can't think of any other case where culling wouldn't be wanted.
Additionally, it might make sense to make the default (or the only) value be the width of the polyline - if that's possible, I realise we're at the layer level here - or something like 10, just because we don't want to introduce visible capping of the polylines at the edges of the screen (the capping needs to overflow off screen).

Reply:
the culling already does not affect gradients! I also cannot think of a use case where polyline culling would want to exist. i have only kept it to prevent an api breaking change.

we do need to keep polyline culling margin since i cannot calculate the lat long radius for meter polygons easily. otherwise we could infer it from width. but since simplification is based on lat longs right now it isn't really easy to handle. that is why i want to eventually change it to be pixel based like leaflet (not just for the sake of copying them, but i think that pixel based simplification just makes more sense).

Reply:
Ok, in that case, as there's so many breaking changes going on elsewhere anyway (and I can't remember if you changed the default, but that would be breaking), let's remove the toggle.

I see what you're saying about the margin. For now then, let's just make it something sensible but non-zero, just so that there aren't any visual glitches, unless you think that's really not necessary.

Reply:
i think its okay to be zero for a default value, most lines aren't thick enough to be problematic (and if it is that is when you should be using the value anyways)